streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorTopology.java

```diff
@@ -59,4 +78,78 @@ public SourceNode source(String topic) {
         return stateStoreSuppliers;
     }
 
+
+    /**
+     * Produces a string representation contain useful information about a ProcessorTopology.
+     * This is useful in debugging scenarios. Prints graph by performing a breadth-first search starting
+     * from source nodes
+     * @return A string representation of the ProcessorTopology instance.
+     */
+    public String toString() {
+        Map<String, Boolean> visitedNode = new HashMap<String, Boolean>();
+        Queue queue = new LinkedList();
+        String processorTopologyString = "ProcessorTopology[";
+        Set<SourceNode> tmpSources = sources();
+
+        // pre-process source nodes to get reverse mapping
+        Map<String, String> sourceNameToTopic = new HashMap<>();
+        for (String topic : sourceByTopics.keySet()) {
+            SourceNode source = sourceByTopics.get(topic);
+            sourceNameToTopic.put(source.name(), topic);
+        }
+
+        // pre-process sink nodes to get reverse mapping
+        Map<String, String> sinkNameToTopic = new HashMap<>();
+        for (String topic : sinkByTopics.keySet()) {
+            SinkNode sink = sinkByTopics.get(topic);
+            sinkNameToTopic.put(sink.name(), topic);
+        }
+
+        // add all sources to a queue for Breadth-First traversal
+        if (tmpSources != null) {
+            // add all sources
+            for (String topic : sourceByTopics.keySet()) {
+                SourceNode source = sourceByTopics.get(topic);
+                // mark as visited
+                visitedNode.put(source.name(), true);
+                // put in queue
+                queue.add(source);
+            }
+
+            // iterate to next level
```

Comment:
I feel the following printing will loose some edge information, for example say you have a three-node topology where both source `a` and `b` links to `c`. This will be printed as

```
a --> c -->
b -->
```

In this case we do not know if c is b's child or not, right?